src/server/cluster/cluster_family.cc

```diff
@@ -712,15 +713,56 @@ void ClusterFamily::DflyClusterMigrationFinalize(CmdArgList args, ConnectionCont
     return cntx->SendError("Migration process is not in STABLE_SYNC state");
   }
 
-  shard_set->pool()->AwaitFiberOnAll([migration](auto*) {
-    if (const auto* shard = EngineShard::tlocal(); shard)
-      migration->Finalize(shard->shard_id());
+  const auto deleted_slots = ToSlotSet(migration->GetSlotRange());
+
+  shard_set->RunBriefInParallel(
+      [](EngineShard* shard) { shard->db_slice().SetExpireAllowed(false); });
+  VLOG(2) << "Disable expiration";
+
+  server_family_->service().proactor_pool().AwaitFiberOnAll(
+      [&deleted_slots](auto*) { ServerState::tlocal()->SetMigratedSlots(deleted_slots); });
+
```

Comment:
1. server_family_service_proactor_pool should be the same as shard_set
2. Why do need two dispatches? Don't you set just thread local stuff?
3. SetMigratedSlots doesn't await anything, so it can be a RunBrief